src/main/java/bio/terra/cli/service/utils/SamService.java

```diff
@@ -366,7 +369,28 @@ public void addUserToResource(
           },
           SamService::isRetryable);
     } catch (ApiException | InterruptedException ex) {
-      throw new SystemException("Error adding user to SAM resource.", ex);
+      if (!(ex instanceof ApiException)
+          || (((ApiException) ex).getCode() != HttpStatusCodes.STATUS_CODE_BAD_REQUEST)) {
+        throw new SystemException("Error adding user to SAM resource.", ex);
+      }
+      logger.info("User not found in SAM. Trying to invite a new user.");
+
+      try {
+        // add to resource failed with Not Found error, now try to invite the user and add them to
+        // the resource again
```

Comment:
looks like it's bad request, not "not found" for this invite the user case.

Reply:
Fixed.